imports/plugins/core/accounts/client/components/email/updateEmail.js

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Meteor } from "meteor/meteor";
+import { i18next } from "/client/api";
+import { Components } from "@reactioncommerce/reaction-components";
+
+class UpdateEmail extends Component {
+  static propTypes = {
+    email: PropTypes.string,
+    uniqueId: PropTypes.string
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: props.email,
+      newEmail: false,
+      showSpinner: false
+    };
+
+    this.handleFieldChange = this.handleFieldChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentWillReceiveProps() {
+    this.setState({ showSpinner: false });
+  }
+
+  emailUpdate() {
+    if (this.state.email === this.props.email) {
+      return false;
+    }
+
+    return true;
+  }
+
+  handleFieldChange = (event, value, field) => {
+    this.setState({
+      [field]: value
+    });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault;
+
+    this.setState({ showSpinner: true });
+
+    const { email } = this.state;
+
+    Meteor.call("accounts/validation/email", email, false, (result, error) => {
+      if (error.error) {
```

Comment:
All server communication or data fetching should always be in a container.  This should just be passed in as an event handler prop.  The UI component shouldn't need to know/care about how something gets done.  It just handles the state of the UI.